pkg/provisioner/controller.go

```diff
@@ -638,3 +656,48 @@ func (c *obcController) objectBucketForClaimKey(key string) (*v1alpha1.ObjectBuc
 	}
 	return ob, nil
 }
+
+func (c *obcController) handleUpdateClaim(key string, obc *v1alpha1.ObjectBucketClaim, class *storagev1.StorageClass) error {
+	ob, err := getObjectBucketFromClaimKey(key, obc, c.libClientset, defaultRetryBaseInterval, defaultRetryTimeout)
+	if err != nil {
+		log.Error(err, "getting ob failed", "key", key)
+		return err
+	}
+	if reflect.DeepEqual(ob.Spec.Endpoint.AdditionalConfigData, obc.Spec.AdditionalConfig) {
+		return nil
+	} else {
+		logD.Info("Additional config changed, hence updating OB")
+		tmp := make(map[string]string)
+
+		copyAdditionalConfig(ob.Spec.Endpoint.AdditionalConfigData, tmp)
+		clearAdditionalConfig(ob.Spec.Endpoint.AdditionalConfigData)
+		copyAdditionalConfig(obc.Spec.AdditionalConfig, ob.Spec.Endpoint.AdditionalConfigData)
```

Comment:
What if, instead of calling `clear...` then `copy...` (which is more of a merge by how it is implemented) there was a function to `overwriteAdditionalConfig` that did both clear and copy (merge) steps? I think that will make it less likely in the future that someone forgets to call `clear` before `copy` when they should do so.

After further thought, It might be more clear to read the functions at a glance as below, which removes the need for `clearAdditionalConfig()` entirely and makes it clear what is being copied and where it is being copied into:
```go
tmp := copyAdditionalConfig(ob.Spec.Endpoint.AdditionalConfigData) // copy into tmp
// copy into (and overwrite) OB endpoint's additional config
ob.Spec.Endpoint.AdditionalConfigData = copyAdditionalConfig(obc.Spec.AdditionalConfig) 

func copyAdditionalConfig(a map[string]string) map[string]string {
    ret := map[string]string{}
    for k, v := range a {
        ret[k] = v
    }
    return ret
}
```